packages/app/src/systems/DApp/pages/SignatureRequest/SignatureRequest.tsx

```diff
@@ -1,15 +1,37 @@
 import { Box, Button, Card, Flex, HelperIcon, Text } from '@fuel-ui/react';
+import type { HashableMessage } from 'fuels';
+import { arrayify } from 'fuels';
 import { AccountInfo } from '~/systems/Account';
 import { ConnectInfo, Layout, coreStyles } from '~/systems/Core';
 
 import { useSignatureRequest } from '../../hooks';
 
+function formatMessage(message: HashableMessage): string {
+  if (typeof message === 'string') {
+    return message;
+  }
+  if (message.personalSign) {
+    try {
+      const bytes = arrayify(message.personalSign);
+      const jsonStr = new TextDecoder().decode(bytes);
+      const jsonObj = JSON.parse(jsonStr);
+      return JSON.stringify(jsonObj, null, 2);
+    } catch (e) {
+      console.error('Error parsing JSON:', e);
+      return 'Invalid JSON format';
+    }
+  }
+  return 'Invalid message format';
+}
```

Comment:
the error feedback should make the sign button disabled, and also show like an error